ImGuiFileDialog.cpp

```diff
@@ -75,300 +103,330 @@ ImGuiFileDialog::~ImGuiFileDialog()
+                        ResetBuffer(FileNameBuffer);
+                        AppendToBuffer(FileNameBuffer, MAX_FILE_DIALOG_NAME_BUFFER, m_SelectedFileName);
+                    }
+                    break;
+                }
+            }
+        }
+
+        // changement de repertoire
+        if (pathClick == true)
+        {
+            ScanDir(m_CurrentPath);
+            m_CurrentPath_Decomposition = splitStringVector(m_CurrentPath, PATH_SEP);
+            if (m_CurrentPath_Decomposition.size() == 2)
+                if (m_CurrentPath_Decomposition[1] == "")
+                    m_CurrentPath_Decomposition.erase(m_CurrentPath_Decomposition.end() - 1);
+        }
+
+        ImGui::EndChild();
+
+        ImGui::Text("File Name : ");
+
+        ImGui::SameLine();
+
+        float width = ImGui::GetContentRegionAvailWidth();
+        if (vFilters != 0) width -= 120.0f;
+        ImGui::PushItemWidth(width);
+        ImGui::InputText("##FileName", FileNameBuffer, MAX_FILE_DIALOG_NAME_BUFFER);
+        ImGui::PopItemWidth();
+
+        if (vFilters != 0)
+        {
+            ImGui::SameLine();
+
+            ImGui::PushItemWidth(100.0f);
+            bool comboClick = ImGui::Combo("##Filters", &FilterIndex, vFilters) || m_CurrentFilterExt.size() == 0;
+            ImGui::PopItemWidth();
+            if (comboClick == true)
+            {
+                int itemIdx = 0;
+                const char* p = vFilters;
+                while (*p)
+                {
+                    if (FilterIndex == itemIdx)
+                    {
+                        m_CurrentFilterExt = std::string(p);
+                        break;
+                    }
+                    p += strlen(p) + 1;
+                    itemIdx++;
+                }
+            }
+        }
+
+        if (ImGui::Button("Cancel"))
+        {
+            IsOk = false;
+            res = true;
+        }
+
+        ImGui::SameLine();
+
+        if (ImGui::Button("Ok"))
+        {
+            if ('\0' != FileNameBuffer[0])
+            {
+
+
+
+                IsOk = true;
+                res = true;
+            }
+        }
+
+        ImGui::EndPopup();
+    }
+
+    if (res == true)
+    {
+        m_FileList.clear();
+    }
+
+    return res;
+}
+
+void ImGuiFileDialog::Reset()
+{
+    m_FileList.clear();
+    m_SelectedFileName.clear();
+    m_CurrentPath.clear();
+    m_CurrentPath_Decomposition.clear();
 }
 
 std::string ImGuiFileDialog::GetFilepathName()
 {
-	return GetCurrentPath() + "\\" + GetCurrentFileName();
+    std::string  result = m_CurrentPath;
+
+    if (s_fs_root != result)
+    {
+        result += PATH_SEP;
+    }
+
+    result += GetCurrentFileName();
+
+    return result;
 }
 
 std::string ImGuiFileDialog::GetCurrentPath()
 {
-	return m_CurrentPath;
+    return m_CurrentPath;
 }
 
 std::string ImGuiFileDialog::GetCurrentFileName()
 {
-	return std::string(FileNameBuffer);
+    std::string result = FileNameBuffer;
```

Comment:
If the user hasn't typed the extension into the filename, append it. This prevents a fixed extension type being chosen but then the user types a different extension. For Windows which is case insensitive, the check if the extension is already present should be made case insensitive.